packages/eslint-plugin/tests/rules/no-unused-vars/no-unused-vars.test.ts

```diff
@@ -1072,6 +1072,12 @@ export class Foo {
         typescript: '4.4',
       },
     },
+    `
+interface Foo {
+  bar: string;
+}
+export const Foo = 'bar';
+    `,
```

Comment:
[Testing] I peeked through the existing tests and couldn't find one that verifies this still gives an error when the variable _isn't_ exported. Am I missing it? If not, could you add one in please? (basically this without the `export`)

Reply:
Ah, good spot.
Even though we already have tests for 
```
interface Foo {
  bar: string;
  baz: Foo['bar'];
}
```
and 
```
interface Foo {
  a: string;
}
interface Foo {
  b: Foo;
}
```
there were no tests for interface + const so I've added one. 

Also since I'm here, added a case for when `export const` is before `interface` to ensure that we are checking all the definitions and not only the last one.

Reply:
Ah great, thanks!